app/controllers/api/v1/surveys_controller.rb

```diff
@@ -33,6 +43,15 @@ def create
 
   # PATCH/PUT /surveys/1 or /surveys/1.json
   def update
+    @survey.update(survey_params)
+    if @survey.isLive
+      @survey.wentLiveAt = DateTime.now()
+    end
+    if @survey.save
```

Comment:
update already calls save, so you don't need to add this. https://apidock.com/rails/ActiveRecord/Base/update/class